src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt

```diff
@@ -118,7 +122,16 @@ data class PrisonerSchedule(
   val event: String,
   val startTime: LocalDateTime,
   val endTime: LocalDateTime?,
-)
+) {
+  fun isAppointmentType(appointmentType: SupportedAppointmentTypes.Type) = event == appointmentType.code
+
+  fun isTimesAreTheSame(appointment: PrisonAppointment) =
+    startTime == appointment.appointmentDate.atTime(appointment.startTime) &&
+      appointment.appointmentDate.atTime(appointment.endTime) == endTime
```

Comment:
Duplicated functions? Though acting on different objects..

Reply:
Yarp, overloaded due to supporting different types.  Using an interface wouldn't be right here as this is DTO for prison API.